ec/resources/waf/managed_rule.go

```diff
@@ -0,0 +1,480 @@
+			"rule_target_update": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"is_negated": {
+							Type:        schema.TypeBool,
+							Description: "Determines whether the current target, as defined within this object, will be ignored when identifying threats.",
+							Optional:    true,
+						},
+						"is_regex": {
+							Type:        schema.TypeBool,
+							Description: "Identifies a rule that will be disabled by its system-defined ID.",
+							Required:    true,
+						},
+						"replace_target": {
+							Type:        schema.TypeString,
+							Description: "Defines the data source (e.g., REQUEST_COOKIES, ARGS, GEO, etc.) that will be used instead of the one defined in the target parameter.",
+							Optional:    true,
+						},
+						"rule_id": {
+							Type:        schema.TypeString,
+							Description: "Identifies a rule by its system-defined ID.",
+							Required:    true,
+						},
+						"target": {
+							Type:        schema.TypeString,
+							Description: "Identifies the type of data source (e.g., REQUEST_COOKIES, ARGS, GEO, etc.) for which a target will be created.",
+							Required:    true,
+						},
+						"target_match": {
+							Type:        schema.TypeString,
+							Description: "Identifies a name or category (e.g., cookie name, query string name, country code, etc.) for the data source defined in the target parameter.",
+							Required:    true,
+						},
+					},
+				},
+				Description: "This array identifies each rule that has been disabled using these properties",
+			},
+		},
+	}
+}
+
+func ResourceManagedRuleCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	accountNumber := d.Get("account_number").(string)
+
+	log.Printf("[INFO] Creating WAF Managed Rule for Account >> %s", accountNumber)
+
+	managedRuleRequest := sdkwaf.AddManagedRuleRequest{}
+
+	managedRuleRequest.Name = d.Get("name").(string)
+	managedRuleRequest.RulesetID = d.Get("ruleset_id").(string)
+	managedRuleRequest.RulesetVersion = d.Get("ruleset_version").(string)
+
+	if policies, ok := helper.ConvertInterfaceToStringArray(d.Get("policies")); ok {
+		managedRuleRequest.Policies = *policies
+	} else {
+		return diag.Errorf("Error reading 'policies''")
+	}
+
+	// Disabled Rules
+	disabledRules, err := ExpandDisabledRules(d.Get("disabled_rule"))
+
+	if err != nil {
+		return diag.Errorf("error parsing disabled_rule: %+v", err)
+	}
+
+	managedRuleRequest.DisabledRules = *disabledRules
+
+	// Rule Target Updates
+	ruleTargetUpdates, err := ExpandRuleTargetUpdates(d.Get("rule_target_update"))
+
+	if err != nil {
+		return diag.Errorf("error parsing rule_target_update: %+v", err)
+	}
+
+	managedRuleRequest.RuleTargetUpdates = *ruleTargetUpdates
+
+	// General Settings
+
+	if v, ok := d.GetOk("general_settings"); ok {
+		if generalSettings, err := ConvertInterfaceToGeneralSettings(v); err == nil {
```

Comment:
To be consistent with naming, we should name that method `ExpandGeneralSettings` - we gotta go back and do that renaming in other places too
- "Expand": Terraform format => API Model
- "Flatten" API Model => Terraform format

Reply:
@stevenpaz Totally clear now, thanks again. corrected.